client/internal/clientcommon.go

```diff
@@ -54,6 +54,9 @@ type ClientCommon struct {
 
 	// Indicates that the Client is fully stopped.
 	stoppedSignal chan struct{}
+
+	// Mutex for Package Syncing
+	PackageSyncComplete sync.Mutex
```

Comment:
Nit: we use "Mutex" prefix elsewhere in this codebase.
```suggestion
	PackageSyncMutex sync.Mutex
```